src/main/scala/piconot/external/IR.scala

```diff
@@ -0,0 +1,82 @@
+package piconot.external
+import picolib.semantics._
+
+/*
+ * Intermediate representations used to simplify the parser and
+ * potentially make it more extensible. 
+ */
+sealed abstract class ExternalDSL
+
+// A list of of the "Open" directions in Surroundings
+case class Free (freeList: List[MoveDirection]) extends ExternalDSL{ 
+  def toFreeArray = {
+    freeList.toArray
+  } 
+}
+
+// A list of the "Blocked" directions in Surroundings
+case class Occupied (occList: List[MoveDirection]) extends ExternalDSL{ 
+  def toOccArray = {
+    occList.toArray
+  } 
+}
+
+// Representation of all of the elements needed to formulate a Rule
+case class Condition (currState: Int, free: Free, occ: Occupied, dir: MoveDirection, nextState: Int) 
+  extends ExternalDSL {
+  
+  // Create Surroundings object
+  def setSurroundings (freeArray:Array[MoveDirection], occArray:Array[MoveDirection]) : Surroundings = {     
+    // Define surroundings
+     var surr_array = new Array[RelativeDescription](4)
+     
+    // Initialize all indices to Anything
+     var x = 0
+     for (x<- 0 to (surr_array.size - 1)) {
+       surr_array.update(x, Anything)
+     }
+     
+    // Set N, E, W, or S to Open
+     if (freeArray.contains(North)) {
```

Comment:
This seems like it could be handled with a for loop and a conversion function (or even an implicit conversion)